src/Workspaces/Core/Portable/Options/IDocumentOptions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    /// <summary>
+    /// Returned from a <see cref="IDocumentOptionsProvider"/>
+    /// </summary>
+    interface IDocumentOptions
+    {
+        bool TryGetDocumentOption(Document document, OptionKey option, out object value);
```

Comment:
This is only called by the core options infrastructure, at the point it's no longer generic anyways. It doesn't change anything.